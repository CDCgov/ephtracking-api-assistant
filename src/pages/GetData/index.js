import React, { Component, Fragment } from 'react';

import CodeBlock from '../../components/CodeBlock';
import CIM from '../../components/CIM';
import StateFIPS from '../../containers/StateFIPS';
import SelectYear from '../../containers/SelectYear';
import Preview from '../../components/Preview';

import './GetData.css';

class GetYears extends Component {
  constructor(props) {
    super(props);
    this.state = {
      measureId: null,
      states: [],
      includeCounty: false,
      isSmoothed: false,
      year: null
    };
    this.setMeasureId = this.setMeasureId.bind(this);
    this.handleStateSelect = this.handleStateSelect.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }
  
  setMeasureId(measureId) {
    this.setState({
      measureId,
      states: [],
      year: null
     });
  }

  handleStateSelect(check) {
    this.setState({
      states: check
    });
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;
    this.setState({
      [name]: value
    });
  }

  render() {
    const { measureId, states, includeCounty, isSmoothed, year } = this.state;
    const isValidUrl = measureId && states && states.length > 0;
    const url = `https://ephtracking.cdc.gov/apigateway/api/v1/getData/${measureId}/${states.toString()}/${includeCounty ? 1 : 0}/${year}/${isSmoothed ? 1 : 0}/json`
    return (
      <Fragment>
        <h1 className="title">Retrieving Data Values for a Measure</h1>
        <h5 className="title is-5">Usage</h5>
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/{'{'}version{'}'}/getData/{'{'}measureId{'}'}/{'{'}state{'}'}/{'{'}includeCounty{'}'}/{'{'}temporal{'}'}/{'{'}isSmoothed{'}'}/{'{'}returnType{'}'}[?apiToken]
        </CodeBlock>
        <hr />
        <h5 className="title is-5">Set parameters</h5>
        <CIM handleSelect={this.setMeasureId} />
        <StateFIPS
          measureId={this.state.measureId}
          handleCheck={this.handleStateSelect}
        />
        <div className="field">
          <label className="checkbox">
            <input name="includeCounty" type="checkbox" checked={this.state.includeCounty} onChange={this.handleInputChange} className="input-checkbox" />
            Include County
          </label>
        </div>
        <SelectYear measureId={this.state.measureId} handleSelect={year => this.setState({ year })} />
        <div className="field">
          <label className="checkbox">
            <input name="isSmoothed" type="checkbox" checked={this.state.isSmoothed} onChange={this.handleInputChange} className="input-checkbox" />
            Is Smoothed
          </label>
          <p className="help">Note: the majority of measures do not have smoothing values</p>
        </div>
        <hr />
        <Preview url={url} validUrl={isValidUrl} />
      </Fragment>
    );
  }
}

export default GetYears;
