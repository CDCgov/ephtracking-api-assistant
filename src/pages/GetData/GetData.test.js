import React from 'react';
import { shallow } from 'enzyme';

import GetData from './';

describe('Page GetData', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GetData />);
    expect(wrapper).toMatchSnapshot();
  });

  it('sets measureId correctly', () => {
    const wrapper = shallow(<GetData />);
    wrapper.instance().setMeasureId('123');
    expect(wrapper.state().measureId).toBe('123');
  });

  it('handles stateSelect correctly', () => {
    const wrapper = shallow(<GetData />);
    wrapper.instance().handleStateSelect(['01', '02', '04']);
    expect(wrapper.state().states).toEqual(['01', '02', '04']);
  });

  it('handles year select correctly', () => {
    const wrapper = shallow(<GetData />);
    wrapper.instance().handleYearSelect('2012');
    expect(wrapper.state().year).toEqual('2012');
  });


  it('can handle input change', () => {
    const wrapper = shallow(<GetData />);
    const event = {
      target: {
        name: 'getChildMeasure',
        checked: true,
        type: 'checkbox'
      }
    };
    wrapper.instance().handleInputChange(event);
    expect(wrapper.state().getChildMeasure).toBe(true);
  });
});
