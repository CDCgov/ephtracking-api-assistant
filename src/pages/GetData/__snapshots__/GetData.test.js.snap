// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page GetData renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GetData />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Retrieving Data Values for a Measure
</h1>,
        <h5
          className="title is-5"
>
          Usage
</h5>,
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/
          {
          version
          }
          /getData/
          {
          measureId
          }
          /
          {
          state
          }
          /
          {
          includeCounty
          }
          /
          {
          temporal
          }
          /
          {
          isSmoothed
          }
          /
          {
          returnType
          }
          [?apiToken]
</CodeBlock>,
        <hr />,
        <h5
          className="title is-5"
>
          Set parameters
</h5>,
        <CIM
          handleSelect={[Function]}
/>,
        <StateFIPS
          handleCheck={[Function]}
          measureId={null}
/>,
        <div
          className="field"
>
          <label
                    className="checkbox"
          >
                    <input
                              checked={false}
                              className="input-checkbox"
                              name="includeCounty"
                              onChange={[Function]}
                              type="checkbox"
                    />
                    Include County
          </label>
</div>,
        <SelectYear
          handleSelect={[Function]}
          measureId={null}
/>,
        <div
          className="field"
>
          <label
                    className="checkbox"
          >
                    <input
                              checked={false}
                              className="input-checkbox"
                              name="isSmoothed"
                              onChange={[Function]}
                              type="checkbox"
                    />
                    Is Smoothed
          </label>
          <p
                    className="help"
          >
                    Note: the majority of measures do not have smoothing values
          </p>
</div>,
        <hr />,
        <Preview
          url="https://ephtracking.cdc.gov/apigateway/api/v1/getData/null//0/null/0/json"
          validUrl={null}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retrieving Data Values for a Measure",
          "className": "title",
        },
        "ref": null,
        "rendered": "Retrieving Data Values for a Measure",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Usage",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Usage",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/getData/",
            "{",
            "measureId",
            "}",
            "/",
            "{",
            "state",
            "}",
            "/",
            "{",
            "includeCounty",
            "}",
            "/",
            "{",
            "temporal",
            "}",
            "/",
            "{",
            "isSmoothed",
            "}",
            "/",
            "{",
            "returnType",
            "}",
            "[?apiToken]",
          ],
        },
        "ref": null,
        "rendered": Array [
          "https://ephtracking.cdc.gov/apigateway/api/",
          "{",
          "version",
          "}",
          "/getData/",
          "{",
          "measureId",
          "}",
          "/",
          "{",
          "state",
          "}",
          "/",
          "{",
          "includeCounty",
          "}",
          "/",
          "{",
          "temporal",
          "}",
          "/",
          "{",
          "isSmoothed",
          "}",
          "/",
          "{",
          "returnType",
          "}",
          "[?apiToken]",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Set parameters",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Set parameters",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSelect": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleCheck": [Function],
          "measureId": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className="checkbox"
>
            <input
                        checked={false}
                        className="input-checkbox"
                        name="includeCounty"
                        onChange={[Function]}
                        type="checkbox"
            />
            Include County
</label>,
          "className": "field",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                className="input-checkbox"
                name="includeCounty"
                onChange={[Function]}
                type="checkbox"
/>,
              "Include County",
            ],
            "className": "checkbox",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "className": "input-checkbox",
                "name": "includeCounty",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "Include County",
          ],
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSelect": [Function],
          "measureId": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="checkbox"
>
              <input
                            checked={false}
                            className="input-checkbox"
                            name="isSmoothed"
                            onChange={[Function]}
                            type="checkbox"
              />
              Is Smoothed
</label>,
            <p
              className="help"
>
              Note: the majority of measures do not have smoothing values
</p>,
          ],
          "className": "field",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="input-checkbox"
                  name="isSmoothed"
                  onChange={[Function]}
                  type="checkbox"
/>,
                "Is Smoothed",
              ],
              "className": "checkbox",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "input-checkbox",
                  "name": "isSmoothed",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "Is Smoothed",
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Note: the majority of measures do not have smoothing values",
              "className": "help",
            },
            "ref": null,
            "rendered": "Note: the majority of measures do not have smoothing values",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://ephtracking.cdc.gov/apigateway/api/v1/getData/null//0/null/0/json",
          "validUrl": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Retrieving Data Values for a Measure
</h1>,
          <h5
            className="title is-5"
>
            Usage
</h5>,
          <CodeBlock>
            https://ephtracking.cdc.gov/apigateway/api/
            {
            version
            }
            /getData/
            {
            measureId
            }
            /
            {
            state
            }
            /
            {
            includeCounty
            }
            /
            {
            temporal
            }
            /
            {
            isSmoothed
            }
            /
            {
            returnType
            }
            [?apiToken]
</CodeBlock>,
          <hr />,
          <h5
            className="title is-5"
>
            Set parameters
</h5>,
          <CIM
            handleSelect={[Function]}
/>,
          <StateFIPS
            handleCheck={[Function]}
            measureId={null}
/>,
          <div
            className="field"
>
            <label
                        className="checkbox"
            >
                        <input
                                    checked={false}
                                    className="input-checkbox"
                                    name="includeCounty"
                                    onChange={[Function]}
                                    type="checkbox"
                        />
                        Include County
            </label>
</div>,
          <SelectYear
            handleSelect={[Function]}
            measureId={null}
/>,
          <div
            className="field"
>
            <label
                        className="checkbox"
            >
                        <input
                                    checked={false}
                                    className="input-checkbox"
                                    name="isSmoothed"
                                    onChange={[Function]}
                                    type="checkbox"
                        />
                        Is Smoothed
            </label>
            <p
                        className="help"
            >
                        Note: the majority of measures do not have smoothing values
            </p>
</div>,
          <hr />,
          <Preview
            url="https://ephtracking.cdc.gov/apigateway/api/v1/getData/null//0/null/0/json"
            validUrl={null}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retrieving Data Values for a Measure",
            "className": "title",
          },
          "ref": null,
          "rendered": "Retrieving Data Values for a Measure",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Usage",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Usage",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "https://ephtracking.cdc.gov/apigateway/api/",
              "{",
              "version",
              "}",
              "/getData/",
              "{",
              "measureId",
              "}",
              "/",
              "{",
              "state",
              "}",
              "/",
              "{",
              "includeCounty",
              "}",
              "/",
              "{",
              "temporal",
              "}",
              "/",
              "{",
              "isSmoothed",
              "}",
              "/",
              "{",
              "returnType",
              "}",
              "[?apiToken]",
            ],
          },
          "ref": null,
          "rendered": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/getData/",
            "{",
            "measureId",
            "}",
            "/",
            "{",
            "state",
            "}",
            "/",
            "{",
            "includeCounty",
            "}",
            "/",
            "{",
            "temporal",
            "}",
            "/",
            "{",
            "isSmoothed",
            "}",
            "/",
            "{",
            "returnType",
            "}",
            "[?apiToken]",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Set parameters",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Set parameters",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSelect": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleCheck": [Function],
            "measureId": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className="checkbox"
>
              <input
                            checked={false}
                            className="input-checkbox"
                            name="includeCounty"
                            onChange={[Function]}
                            type="checkbox"
              />
              Include County
</label>,
            "className": "field",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="input-checkbox"
                  name="includeCounty"
                  onChange={[Function]}
                  type="checkbox"
/>,
                "Include County",
              ],
              "className": "checkbox",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "input-checkbox",
                  "name": "includeCounty",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "Include County",
            ],
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSelect": [Function],
            "measureId": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="checkbox"
>
                <input
                                checked={false}
                                className="input-checkbox"
                                name="isSmoothed"
                                onChange={[Function]}
                                type="checkbox"
                />
                Is Smoothed
</label>,
              <p
                className="help"
>
                Note: the majority of measures do not have smoothing values
</p>,
            ],
            "className": "field",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    checked={false}
                    className="input-checkbox"
                    name="isSmoothed"
                    onChange={[Function]}
                    type="checkbox"
/>,
                  "Is Smoothed",
                ],
                "className": "checkbox",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "className": "input-checkbox",
                    "name": "isSmoothed",
                    "onChange": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "Is Smoothed",
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Note: the majority of measures do not have smoothing values",
                "className": "help",
              },
              "ref": null,
              "rendered": "Note: the majority of measures do not have smoothing values",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "https://ephtracking.cdc.gov/apigateway/api/v1/getData/null//0/null/0/json",
            "validUrl": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
